lib/schemas.js

```diff
@@ -129,19 +136,37 @@ function getSchemaAnyway (schema, jsonShorthand) {
  *
  * @param {object} context the request context
  * @param {number} statusCode the http status code
+ * @param {string} contentType the reply content type
  * @returns {function|boolean} the right JSON Schema function to serialize
  * the reply or false if it is not set
  */
-function getSchemaSerializer (context, statusCode) {
+function getSchemaSerializer (context, statusCode, contentType) {
   const responseSchemaDef = context[kSchemaResponse]
   if (!responseSchemaDef) {
     return false
   }
   if (responseSchemaDef[statusCode]) {
+    if (responseSchemaDef[statusCode].constructor === Object) {
+      const type = contentType.split(';')[0]
+      if (responseSchemaDef[statusCode][type]) {
+        return responseSchemaDef[statusCode][type]
+      }
+
+      return false
+    }
     return responseSchemaDef[statusCode]
   }
   const fallbackStatusCode = (statusCode + '')[0] + 'xx'
   if (responseSchemaDef[fallbackStatusCode]) {
+    if (responseSchemaDef[fallbackStatusCode].constructor === Object) {
+      const type = contentType.split(';')[0]
+      if (responseSchemaDef[fallbackStatusCode][type]) {
+        return responseSchemaDef[fallbackStatusCode][type]
+      }
+
+      return false
+    }
+
     return responseSchemaDef[fallbackStatusCode]
   }
   if (responseSchemaDef.default) {
```

Comment:
I've just noticed that I didn't handle it in the case of `default` schema being used, what do you think? should the `default` schema support different content types?

Reply:
wdyt @Eomm?